### ISSUE TITLE
Implement intuitive "Add Account" settings options

### ISSUE THREAD

**Issue by bnvk** [opensourcedesign/slick#12]
In Slick, I'm treating the idea of IRC networks as an "Account" instead of a network, as many desktop clients do this, and lends itself to better metaphor (i think).

![screen shot 2015-05-17 at 9 42 48 pm](https://cloud.githubusercontent.com/assets/48677/7671816/f32e09dc-fcdd-11e4-892b-68a831721672.png)

The thinking being that one can "add" an account at anytime with a form like this

![screen shot 2015-05-17 at 9 43 02 pm](https://cloud.githubusercontent.com/assets/48677/7671821/058ae12c-fcde-11e4-9c84-48c21262f738.png)

However, all those form fields are not too user friendly. I think we should offer a dropdown menu (or some sort of intuitive clickable UI element) that allows a user to easily select the network to join.

The popular default networks I'm aware of (and use) are:
- freenode.net
- oftc.net
- pirateirc.net

Please list other popular / useful networks (and their settings) in the comments, and I'll add them :)

**Comment by jdittrich** [comment]
Makes sense. 

As well I wonder if the access to the "network itself" (what your get if you are logged into a network, but not entered a channel) should be different as well. Currently we get the normal chat view, and one can type stuff like `/join #ubuntuusers`but this is hardly intuitive. 

If that makes sense to you – should I open an additional bug for that?

**Comment by simonv3** [comment]
I wonder if this won't be too confusing for people who are semi-familiar with IRC lingo? 

Edit: I have only ever used freenode.

**Comment by bnvk** [comment]
@simonv3 yes, I think these form fields are _definitely_ too confusing for non-IRC familiar folks- thus why I suggest `"I think we should offer a dropdown menu (or some sort of intuitive clickable UI element) that allows a user to easily select the network to join"` :wink:

**Comment by simonv3** [comment]
@bnvk What I meant specifically is changing the terminology from "Network" to "Account" might be confusing if someone who is familiar with how IRC works comes to our IRC room and gets presented with this different terminology.

**Comment by jancborchardt** [comment]
Even for people not familiar with IRC: If we call it »Account« and then directly below (as in the second screenshot) there is a field called »Name« – that is confusing with the username.

**Comment by jancborchardt** [comment]
> as many desktop clients do this

@bnvk do you have examples, and screenshots of how they solve it?

**Comment by bnvk** [comment]
@jancborchardt @simonv3 @jdittrich here is what I'm currently thinking for "Add Account" view

![screen shot 2015-05-19 at 10 02 57 pm](https://cloud.githubusercontent.com/assets/48677/7712653/d5645582-fe72-11e4-9327-c8d88c9383a1.png)

Of course, one _could_ see & edit the details of a network... but tucking it away + adding presets that make it easier is what I'm going for. I'd love to see other peoples ideas about alternate solutions for this UX issue!

**Comment by jancborchardt** [comment]
Some layout / design fixes:
![shout add account](https://cloud.githubusercontent.com/assets/925062/7721437/e980d490-fed7-11e4-85ad-6ebb8fba6d34.png)
(I put the Network _below_ the username and full name because it’s less important.)

I can also directly do it in code. master branch of this repo?

**Comment by bnvk** [comment]
@jancborchardt :+1: to that idea!

**Comment by simonv3** [comment]
An idea: instead, or alongside, of Details have a "what does this mean?" link available? Or would that appear inside details?

**Comment by jdittrich** [comment]
^- Makes sense. "Details" is not very meaningful itself.

**Comment by simonv3** [comment]
And another thought: what happens when a user has a password assigned to the name?

**Comment by jancborchardt** [comment]
> Does details have a "what does this mean?" link available? Or would that appear inside details?

Yep, should have. :)

> What happens when a user has a password assigned to the name?

They get prompted for it. And we could also show it in the »Details« – which we could call »Advanced settings« or whatnot for people who do have a password protected account and already want to put it in in this step.

**Comment by simonv3** [comment]
I like "Advanced Settings". I wonder if we can ping the username as soon as it's entered (and the box blurred) and parse the response for the "that username is password protected", so that based on the username the password box shows up (we can give ourselves a bit more time by creating a spinner). I don't know enough about what kind of response we get back from the IRC backend though, and how easy it is to intercept.
